### PR TITLE
Ensure "Save schedule" button also trigger the overwrite confirmation dialog

### LINKED ISSUE #12158: Concurrent editing notification: overwrite confirmation dialog does not show when saving a publishing schedule
### Issue Summary

With the new concurrent editing notification feature, trying to submit the editor form when there's a newer revision (since the one that was initially loaded) will trigger a confirmation dialog to make the user aware that they are about to overwrite the new revision.

However, this confirmation dialog does not show when you try to set a publishing schedule.

### Steps to Reproduce

1. Spin up bakerydemo
2. Open the edit view of a page in two different tabs
3. Make some edits and save the page on one tab
4. Go to the other tab, ensure the "has saved a new version" notification shows up at the header
5. Try to set a publishing schedule from the status side panel
6. Upon clicking "Save schedule", no confirmation dialog is shown, and you have just overwritten the edits you made in the other tab.

Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?

The "Save schedule" works by saving a new revision that includes the `go_live_at` and `expire_at` fields that were set in the dialog, so it should also have the confirmation dialog guard.

-   I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: no

### Technical details

-   Wagtail version: Latest `main` branch
-   Browser version: Chrome 126, Safari 17.5, Firefox 128

### Working on this

I'll work on this.

### PR DESCRIPTION
**Pull request by laymonage**
Fixes #12158.

### CODE REVIEW

File: client/src/controllers/SessionController.ts [hunk 1]
```diff
@@ -87,14 +87,6 @@ export class SessionController extends Controller<HTMLElement> {
...
 
   connect(): void {
-    this.interceptTargets.forEach((button) => {
-      // Match the event listener configuration of workflow-action that uses
-      // capture so we can intercept the workflow-action's listener.
-      button.addEventListener('click', this.showConfirmationDialog, {
-        capture: true,
-      });
-    });
-
```
Comment (laymonage) [inline client/src/controllers/SessionController.ts:97]: The problem was, this controller is connected before the schedule publishing dialog is teleported to the edit form. As a result, `querySelectorAll` won't find the submit button yet.

We're fixing this by moving the query to `wDialogOutletConnected`, which is fired when the overwrite changes dialog template gets teleported and the its DialogController is connected. Fortunately, this happens after the schedule publishing dialog template is teleported, so its submit button is included in the query result.

This fix won't work for any submit button added to the editor form after that. We can't reliably detect the addition of those buttons without using MutationObserver on the form element and looking for new submit buttons, but I feel like that's an unnecessary overhead for something that we ourselves don't need.

Custom buttons that are added too late can still be hooked into the mechanism by adding the click event listener with the `SessionController.showConfirmationDialog` as the handler, so I think it's fair to leave it up to them to do this.

File: client/src/controllers/SessionController.ts [hunk 2]
```diff
@@ -197,12 +195,32 @@ export class SessionController extends Controller<HTMLElement> {
...
     );
   }
 
+  wDialogOutletDisconnected(): void {
+    this.interceptTargets?.forEach((button) => {
+      button.removeEventListener('click', this.showConfirmationDialog, {
+        capture: true,
+      });
+    });
+  }
```
Comment (laymonage) [inline client/src/controllers/SessionController.ts:222]: Moving this from `disconnect()` to here for consistency with how we moved the `addEventListener` to `wDialogOutletConnected`

File: client/src/controllers/SessionController.ts [hunk 3]
```diff
@@ -175,6 +167,12 @@ export class SessionController extends Controller<HTMLElement> {
     // workflow action modal) after the user confirms the dialog.
     if (!this.interceptValue || !this.hasWDialogOutlet) return;
 
+    // If the action button is inside a dialog, we need to hide the dialog first
+    // so it doesn't interfere with the confirmation dialog
+    this.lastActionButton
+      ?.closest('[data-controller="w-dialog"]')
+      ?.dispatchEvent(new Event('w-dialog:hide'));
```
Comment (laymonage) [inline client/src/controllers/SessionController.ts:174]: We can also show the dialog again after the overwrite has been confirmed by dispatching the `w-dialog:show` event in `confirmAction`, but I opted not to as per our discussion.

**Review by thibaudcolas — APPROVED** [review]
Thank you!

**Pull request merged**